File: Backend/MoviesWebApplication/Movies.Domain/Director.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.ComponentModel.DataAnnotations;

namespace Movies.Domain
{
    public class Director
    {
        [Key]
        public Guid DirectorId { get; set; }
        public string FirstName { get; set; }
        public string LastName { get; set; }
        public string Bio { get; set; }
        public DateTime BirthDate { get; set; }
        public string PhotoUrl { get; set; }

        //Kolekcja potrzebna do wygenerowania klucza obcego
        public ICollection<Movie> Movies { get; set; }
    }
}

File: Frontend/projekt-filmy-client-app/src/models/Director.ts
export interface Director {
    directorId: string;
    firstName: string;
    lastName: string;
    bio: string;
    birthDate: string;
    photoUrl: string;
    //Dodatkowe pola
    totalMovies: number;
    favoriteGenre: string;
}

export interface DirectorResponse {
    $values: Director[];
}